cdflow_commands/cli.py

```diff
@@ -121,25 +122,38 @@ def _setup_for_infrastructure(
     write_terragrunt_config(
         metadata.aws_region, s3_bucket, environment_name, component_name
     )
-    return boto_session, platform_config_file
+    return boto_session, platform_config_file, s3_bucket
 
 
 def _run_deploy(
     team, platform_config_file, boto_session, component_name, environment_name,
-    version, dev_account_id
+    version, global_config
 ):
+    is_prod = environment_name == 'live'
+    if is_prod:
+        cluster = global_config.prod_ecs_cluster
+    else:
+        cluster = global_config.dev_ecs_cluster
+
     deploy_config = DeployConfig(
         team=team,
-        dev_account_id=dev_account_id,
+        dev_account_id=global_config.dev_account_id,
         platform_config_file=platform_config_file,
     )
     deployment = Deploy(
         boto_session, component_name, environment_name, version,
         deploy_config
     )
     deployment.run()
+    events = ECSEventIterator(
+        cluster, environment_name, component_name, version, boto_session
+    )
+    monitor = ECSMonitor(events)
+    monitor.wait()
```

Comment:
Using a parent exception for both `TimeoutError` and `ImageDoesNotMatchError` could be good as the parent could be caught and output here before exiting with a non-zero code.